src/addIndex.js

```diff
@@ -20,7 +20,7 @@ var curryN = require('./curryN');
  * @category List
  * @sig ((a ... -> b) ... -> [a] -> *) -> (a ..., Int, [a] -> b) ... -> [a] -> *)
```

Comment:
I don't understand the `...` in `(a ..., Int, [a])`. Shouldn't it just be `(a, Int, [a])`?

Can you share your suggestion for the entire signature?